### ISSUE TITLE
Is there a way to receive updates on the device performing relay hops?

### ISSUE THREAD

**Issue by jeffjvick** [gotenna/PublicSDK#37]
Hi, I haven't completely looked at everything in the SDK but on first glance it does not appear there is way to receive status updates from the device when it is performing relay hops for other messages.  What I mean by that is, is there a way to know when you device has just forwarded a message initiated by another goTenna user?  Like, which hop count it was and if the message eventually reached the desired GID.  

Thanks

**Comment by tcolligan-ap** [comment]
You can currently only determine the hop count for messages you have personally sent via the sdk to another user.

Using the SDK, there is currently no way to tell if your device was used as a hopping off point for another person's message.

The firmware on the goTenna obviously knows when this situation occurs, but it does not currently forward this type of information to the mobile SDK over Bluetooth.

This is something we have considered exposing, but there are no immediate plans to implement & expose this functionality as far as I am aware.

**Comment by jeffjvick** [comment]
@tcolligan-ap Thanks for the quick response.

This is unfortunate as it limits a number of interesting application designs.

I feel a feature that would help encourage people to setup up more permanent routers would be to have an idea of how many people they "helped" relay messages for.  Otherwise the thing is just sitting and you don't know if it's A: working, B: to what extent.  People need feedback if they are to be encouraged to do something that doesn't directly benefit themselves.

Speaking of benefiting themselves, also, if there is ever a drive to create some type of incentive program for putting up a relay, as is discussed here: https://meshcommunity.gotenna.com/t/blockchain-gotenna-mesh/2383 then something would need to be added to the SDK to fetch this information from the device.

I doesn't even need to be a real-time push type service.  It could merely be a polling call to the device that returns a log of the last X hops performed and timestamps of when they occurred.  Nice to have would be data on which were successful (if the device even knows that) and GID's involved.  The later might raise some anonymity issues though but would be necessary if some type of incentive model were to be built.

**Comment by camswx** [comment]
@jeffjvick Concur with incentives being important for fixed, always-up nodes. I think it would be awesome to find a way to link in this project with the SDK: https://altheamesh.com/

**Comment by jeffjvick** [comment]
@camswx Yeah I like what Althea mesh is doing as well although they are operating at much higher bandwidth and would thus be easier for them to directly implement blockchain features in their transmissions.  

For goTenna I think it would be fine to just start with some type of "novelty" incentives but until we can receive hop information from relays (or even end-user devices), that would not be possible.

**Comment by limnick** [comment]
Would love to see something like this

I think even something as simple as an incrementing counter of relayed messages may be enough
